cmd/shell/lookupCache.go

```diff
@@ -0,0 +1,35 @@
+package shell
+
+import "sync"
+
+type lookupCache struct {
+	mtx   *sync.RWMutex
+	cache map[string]error
+}
+
+func newLookupCache() *lookupCache {
+	var mtx sync.RWMutex
+	return &lookupCache{&mtx, nil}
+}
+
+func (l *lookupCache) fetch(key string) (exists bool, err error) {
+	l.mtx.Lock()
```

Comment:
Would `RLock()` and `RUnlock()`  be better? I guess it's not necessary locking the map for writing here

Reply:
@danielsuguimoto good catch! indeed had opted for RWMutex exactly because we can have a shared lock for reading.